include/zephyr/drivers/bluetooth.h

```diff
@@ -0,0 +1,253 @@
+	 * default data length parameters. Therefore the host should initiate
+	 * the DLE procedure after connection establishment.
+	 */
+	BT_HCI_QUIRK_NO_AUTO_DLE = BIT(1),
+};
+
+/** Possible values for the 'bus' member of the bt_hci_driver struct */
+enum bt_hci_bus {
+	BT_HCI_BUS_VIRTUAL       = 0,
+	BT_HCI_BUS_USB           = 1,
+	BT_HCI_BUS_PCCARD        = 2,
+	BT_HCI_BUS_UART          = 3,
+	BT_HCI_BUS_RS232         = 4,
+	BT_HCI_BUS_PCI           = 5,
+	BT_HCI_BUS_SDIO          = 6,
+	BT_HCI_BUS_SPI           = 7,
+	BT_HCI_BUS_I2C           = 8,
+	BT_HCI_BUS_IPM           = 9,
+};
+
+#define BT_DT_HCI_QUIRK_OR(node_id, prop, idx) DT_STRING_TOKEN_BY_IDX(node_id, prop, idx)
+#define BT_DT_HCI_QUIRKS_GET(node_id) COND_CODE_1(DT_NODE_HAS_PROP(node_id, bt_hci_quirks), \
+						  (DT_FOREACH_PROP_ELEM_SEP(node_id, \
+									    bt_hci_quirks, \
+									    BT_DT_HCI_QUIRK_OR, \
+									    (|))), \
+						  (0))
+#define BT_DT_HCI_QUIRKS_INST_GET(inst) BT_DT_HCI_QUIRKS_GET(DT_DRV_INST(inst))
+
+#define BT_DT_HCI_NAME_GET(node_id) DT_PROP_OR(node_id, bt_hci_name, "HCI")
+#define BT_DT_HCI_NAME_INST_GET(inst) BT_DT_HCI_NAME_GET(DT_DRV_INST(inst))
+
+#define BT_DT_HCI_BUS_GET(node_id) DT_STRING_TOKEN_OR(node_id, bt_hci_bus, BT_HCI_BUS_VIRTUAL)
+#define BT_DT_HCI_BUS_INST_GET(inst) BT_DT_HCI_BUS_GET(DT_DRV_INST(inst))
+
+typedef int (*bt_hci_recv_t)(const struct device *dev, struct net_buf *buf);
+
+__subsystem struct bt_hci_driver_api {
+	int (*open)(const struct device *dev, bt_hci_recv_t recv);
+	int (*close)(const struct device *dev);
+	int (*send)(const struct device *dev, struct net_buf *buf);
+#if defined(CONFIG_BT_HCI_SETUP)
+	int (*setup)(const struct device *dev,
+		     const struct bt_hci_setup_params *param);
+#endif /* defined(CONFIG_BT_HCI_SETUP) */
+};
+
+/**
+ * @brief Open the HCI transport.
+ *
+ * Opens the HCI transport for operation. This function must not
+ * return until the transport is ready for operation, meaning it
+ * is safe to start calling the send() handler.
+ *
+ * @param dev  HCI device
+ * @param recv Callback to receive data from the controller
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_open(const struct device *dev, bt_hci_recv_t recv)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	return api->open(dev, recv);
+}
+
+/**
+ * @brief Close the HCI transport.
+ *
+ * Closes the HCI transport. This function must not return until the
+ * transport is closed.
+ *
+ * @param dev HCI device
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_close(const struct device *dev)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	if (api->close == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->close(dev);
+}
+
+/**
+ * @brief Send HCI buffer to controller.
+ *
+ * Send an HCI command or ACL data to the controller. The exact
```

Comment:
This API can also be used for ISO, so it would be nice to make it a bit more generic
```suggestion
 * Send an HCI packet to the controller. The exact
```

Reply:
Is the buffer H4 formatted? It would be nice to document that

Reply:
Done